src/Messages/Interactions/TSOutgoingMessage.m

```diff
@@ -110,20 +187,163 @@ - (nullable NSString *)recipientIdentifier
 - (BOOL)shouldStartExpireTimer
 {
     switch (self.messageState) {
-        case TSOutgoingMessageStateSent:
-        case TSOutgoingMessageStateDelivered:
+        case TSOutgoingMessageStateSentToService:
             return self.isExpiringMessage;
         case TSOutgoingMessageStateAttemptingOut:
         case TSOutgoingMessageStateUnsent:
             return NO;
+        case TSOutgoingMessageStateSent_OBSOLETE:
+        case TSOutgoingMessageStateDelivered_OBSOLETE:
+            OWSAssert(0);
+            return self.isExpiringMessage;
+    }
+}
+
+#pragma mark - Update Methods
+
+- (void)applyChangeToSelfAndLatestOutgoingMessage:(YapDatabaseReadWriteTransaction *)transaction
```

Comment:
The reason for this method may not be obvious outside of the context of this PR. Could you add a method comment to explain why it exists?